backend/app/api/handlers/v1/v1_ctrl_auth.go

```diff
@@ -89,6 +90,80 @@ func (ctrl *V1Controller) HandleAuthLogin() errchain.HandlerFunc {
 	}
 }
 
+func (ctrl *V1Controller) HandleSsoHeaderLogin() errchain.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) error {
+		log.Info().Msg("Header SSO Login Attempt")
+		if !ctrl.headerSSOEnabled {
+			return validate.NewRequestError(errors.New("authentication failed. Header SSO is disaled"), http.StatusInternalServerError)			
+		}
+
+		{
+			t := strings.Split(r.RemoteAddr, ":")
+			if t[0] != ctrl.headerSSOAllowedIP {
+				return validate.NewRequestError(errors.New("authentication failed. Header SSO not allowed for this remote IP"), http.StatusInternalServerError)			
+			}
+			log.Info().Msgf("Header SSO Login Attempt allowed from IP '%s'", t[0])
+		}
+
+		email := r.Header.Get(ctrl.headerSSOHeaderEmail)
+
+		if email == "" {
+			return validate.NewRequestError(errors.New("authentication failed. not SSO header found or empty"), http.StatusInternalServerError)
+		}
+
+		// check if a user matching provided email is existing already
+		_, err := ctrl.repo.Users.GetOneEmail(r.Context(), email)
+
+		if err != nil {
+			if !ctrl.headerSSOAutoRegister {
+				return validate.NewRequestError(errors.New("authentication failed. User not found but SSO autoregister is disabled"), http.StatusInternalServerError)
+			}
+			
+			// user not found -> create it
+			// if the name header does not exist then the empty string will be used as name
+			var username = r.Header.Get(ctrl.headerSSOHeaderName)
+			
+			//
+			/* TODO: decide how to handle group information provided by HTTP header
+			// if groups are provided, they will be comma-separated. take only the first group
+			var groups = r.Header.Get("Remote-Groups")
+			var groupArr = strings.Split(groups, ",")
+			groupTok := ""
+			if len(groupArr) > 0 {
+				groupTok = groupArr[0]
+			}
+			*/
```

Comment:
Where are you at with the group support? Currently every one will get their own group right now, correct?

Reply:
Regarding groups, please see my previous comment about it:
https://github.com/hay-kot/homebox/pull/345#issuecomment-1522327160

TLDR:
Currently, it is handled the same way as regular registrations: every user is put into the group "Home".
I don't know if it makes sense to use the (possible) group information provided by the HTTP header because this would mean that different services (Homebox and others) need to agree on a common set of groups (group names). At least according to my understanding.

If you have a different idea how to handle the groups from the header, then I can implement that of course.

Reply:
> Currently, it is handled the same way as regular registrations: every user is put into the group "Home".

That isn't how it's currently handled though, unless I'm mis-understanding what you mean. Homebox is multi-tenant so each group has their own set of items, locations, etc... So if you're using SSO I'd assume you'd want some way for users to all be in the same group. I could see this going two ways,

1) Use a header to specify the group and either Add the user if the group exists, or create the group if it doesn't
2) Default all SSO users to some predefined group name (possibly set via env)

Again, maybe I'm just missing obvious...

Reply:
First, pretty sure I am the confused one here ;)

Ok, I didn't know about this multi-tenant thing, good to know, thanks. Anyway, without understanding what groups exactly do, I tried to handle groups in the SSO-Autoregistration the same way as (I thought) it works for regular user registration.
From what I saw in debugging, when doing a regular registration, then `HandleUserRegistration` is called with an empty group string (`""`) (since you cannot select/specify a group when registering?). Then when `RegisterUser` is called with the empty string as `GroupToken`, it falls back to a default group `Home`:
```
	switch data.GroupToken {
	case "":
		log.Debug().Msg("creating new group")
		creatingGroup = true
		group, err = svc.repos.Groups.GroupCreate(ctx, "Home")
		if err != nil {
			log.Err(err).Msg("Failed to create group")
			return repo.UserOut{}, err
		}
	default:
		log.Debug().Msg("joining existing group")
		token, err = svc.repos.Groups.InvitationGet(ctx, hasher.HashToken(data.GroupToken))
		if err != nil {
			log.Err(err).Msg("Failed to get invitation token")
			return repo.UserOut{}, err
		}
		group = token.Group
	}
```

So, that was as my assessment for it, so I also just used the emptry string as `GroupToken` for SSO-Autoregister. Just out of interest: did I miss something here?

Anyway, I like your second idea to make it configurable via environment variable and will go that way!
> Default all SSO users to some predefined group name (possibly set via env)

Reply:
Sorry, me again. This group thing still confuses me a bit, but after further studying the code, I think I had some insights. Also, I realized what I wrote above is not quite correct. The point is that a group is not identified by its name but by a UUID, right? So, there can be multiple groups with the name "Home" and for every normal user registration such a group is created for the new user (I remember you mentioned that above somewhere, and now I understand it).

> Default all SSO users to some predefined group name (possibly set via env)

Do you mean (similar to regular registration) creating a new group for every SSO auto registration and use this predefined group name (instead of "Home")?

Wouldn't it be sensible to handle it the same way as regular user registration with empty group token? So, just creating a new group named "Home" for every SSO user? That would be pretty consistent then with the regular registration. Basically how it is currently ;)

Reply:
So, I see three options to handle groups when auto-registering an SSO user:
1. Create a new group for every user (similar to regular registration) using a configurable default group name (I think that is what you proposed)
2. Handle it the same way as regular registration: create new group named "Home" for every user
3. Join the new user to an existing group by a configurable group UUID

I am fine with any of these. I am not entirely sure if I understood you correctly, so I would wait for confirmation before going on.

Reply:
Any opinions about this? Maybe we can finish this once this group thing is solved.